build-tests/api-extractor-test-01/etc/api-extractor-test-01.api.md

```diff
@@ -49,11 +49,11 @@ export class ClassWithAccessModifiers {
 // @public (undocumented)
 export class ClassWithSymbols {
```

Comment:
Could you add a test case with a nested name e.g. `[X.Y.z]`?  You have a somewhat interesting loop that handles this nesting.

Reply:
I'm actually going to replace this loop with a call to the TypeScript node printer so that it can handle the logic instead.

Reply:
I'm adding a test, but I noticed an issue with the .d.ts rollup WRT namespaces that export custom symbols. It seems like the .d.ts rollup reorders elements, which is causing an error.

In my test I have added the following:

```ts
/** @public */
export namespace Namespace {
  export const locallyExportedCustomSymbol: unique symbol = 
    Symbol('locallyExportedCustomSymbol');

  /** @public */
  export const fullyExportedCustomSymbol: unique symbol = 
    Symbol('fullyExportedCustomSymbol');
}

/**
 * @public
 */
export class ClassWithSymbols {

  …

  public get [Namespace.locallyExportedCustomSymbol](): string {
    return 'hello';
  }

  public [Namespace.fullyExportedCustomSymbol](): void {
  }
}
```

However, in the rollup .d.ts file, it seems to put `Namespace` *after* `ClassWithSymbols`, resulting in the following error when consuming the .d.ts:

```
================================
api-extractor-test-02 (7.47 seconds)
Warning: C:/dev/web-build-tools/build-tests/api-extractor-test-01/dist/
api-extractor-test-01.d.ts:114:25 - (TS2729) Property 'locallyExportedCustomSymbol' is used 
before its initialization.
================================
```

Reply:
Ah, I guess it is because it is alphabetizing the exports. For now I'll just change the name of the namespace so that it comes sooner. However, you may want to keep in mind that sorting the output can cause problems like this.

Reply:
Or perhaps TypeScript should ignore use-before-def errors in .d.ts files. I can file a bug for that.